runtime/jcl/common/stdinit.c

```diff
@@ -576,6 +576,10 @@ initializeBootstrapClassPath(J9JavaVM *vm)
 	if (-1 == (IDATA)loader->classPathEntryCount) {
 		return -1;
 	} else {
+		if (NULL != loader->classPathEntries) {
+			/* classLoader->cpEntriesMutex is initialized only when there are classPathEntries */
+			omrthread_monitor_init_with_name(&loader->cpEntriesMutex, 0, "classPathEntries Mutex");
```

Comment:
If this is always initialized for the boot loader, why do some places (but not all places) check for it non-null after checking for the boot loader? Seems that could mask a bug.

Reply:
In Java 11 and up, it is not always initialized for bootstrap class loader. There is module path but we don't have bootstrap class path by default.

Reply:
Does that mean the places that don't check for NULL are Java 8 specific?

Reply:
They are GetClassPathEntryType, I guess you need to have an entry before checking it's type.

Reply:
The code in GetClassPathEntryType() is protected by 
`if ((cpIndex >= 0) && (cpIndex < (jint)classLoader->classPathEntryCount)) {`

classLoader->classPathEntryCount > 0, so there are class path entries.